Readme.md

```diff
@@ -110,46 +123,47 @@ my-project
 
 ## Python metadata
 
-To specify python dependencies, add a list `requires-dist` in a `[package.metadata.maturin]` section in the Cargo.toml. This list is equivalent to `install_requires` in setuptools:
+maturin supports [PEP 621](https://www.python.org/dev/peps/pep-0621/), you can specify python package metadata in `pyproject.toml`.
+
+To specify python dependencies, add a list `dependencies` in a `[project]` section in the `pyproject.toml`. This list is equivalent to `install_requires` in setuptools:
 
 ```toml
-[package.metadata.maturin]
-requires-dist = ["flask~=1.1.0", "toml==0.10.0"]
+[project]
+dependencies = ["flask~=1.1.0", "toml==0.10.0"]
 ```
 
-Pip allows adding so called console scripts, which are shell commands that execute some function in you program. You can add console scripts in a section `[package.metadata.maturin.scripts]`. The keys are the script names while the values are the path to the function in the format `some.module.path:class.function`, where the `class` part is optional. The function is called with no arguments. Example:
+Pip allows adding so called console scripts, which are shell commands that execute some function in you program. You can add console scripts in a section `[project.scripts]`.
+The keys are the script names while the values are the path to the function in the format `some.module.path:class.function`, where the `class` part is optional. The function is called with no arguments. Example:
 
 ```toml
-[package.metadata.maturin.scripts]
+[project.scripts]
 get_42 = "my_project:DummyClass.get_42"
 ```
 
-You can also specify [trove classifiers](https://pypi.org/classifiers/) in your Cargo.toml under `package.metadata.maturin.classifiers`:
+You can also specify [trove classifiers](https://pypi.org/classifiers/) in your Cargo.toml under `project.classifiers`:
 
 ```toml
-[package.metadata.maturin]
+[project]
 classifiers = ["Programming Language :: Python"]
 ```
 
-Note that `package.metadata.maturin.classifier` is also supported for backwards compatibility.
-
-You can use other fields from the [python core metadata](https://packaging.python.org/specifications/core-metadata/) in the `[package.metadata.maturin]` section, specifically ` maintainer`, `maintainer-email` and `requires-python` (string fields), as well as `requires-external` and `provides-extra` (lists of strings) and `project-url` (dictionary string to string)
-
-## pyproject.toml
+## Source distribution
 
-maturin supports building through pyproject.toml. To use it, create a `pyproject.toml` next to your `Cargo.toml` with the following content:
+maturin supports building source distributions (sdist) through `pyproject.toml`. To use it, create a `pyproject.toml` next to your `Cargo.toml` with the following content:
```

Comment:
PEP 517 is more than just source distributions! [build](https://pypi.org/project/build/) e.g. can build a wheel directly by invoking `build_wheel` with minimal setup and afaik pip has a no isolation mode where it also builds wheels directly (I think https://github.com/pypa/pip/pull/9091 is related). For maturin users this is relevant as they might want to do `pip install .` or `pip wheel .`, which only works with a pyproject.toml .

Reply:
Right. I was thinking the paragraph was only talking about sdist and `pip install`/`pip wheel` usually works with sdist published on PyPI, doesn't work with maturin binary release on GitHub.